### PR TITLE
Ext pillar no recurse

### PR DESCRIPTION
**Pull request by gracinet**
Work in progress, this solves issues #21342 for git_pillar, relying on the test of PR #21675 for that.
But at this point, I'm not sure whether git_pillar is not actually relying on being called back again for some corner cases related to env/branch mapping.

Also I wonder why this `ext` kwarg, introduced in [4d53b498] for seemingly that very same purpose is not actually used with value of `False` in the external pillars that are part of salt's main tree.

Open for comments !

### CODE REVIEW

**Comment by thatch45** [comment]
Very minor lint:
https://jenkins.saltstack.com/job/salt-pr-lint-n/3067/violations/file/tests/unit/pillar/git_test.py/

This test is not passing:
https://jenkins.saltstack.com/job/salt-pr-rs-ubuntu14.04-n/2636/testReport/junit/integration.modules.pillar/PillarModuleTest/test_ext_pillar_env_mapping/

**Comment by jfindlay** [comment]
Go Go Jenkins!

**Comment by jfindlay** [comment]
@gracinet, my guess is that it is an oversight.  Do you consider this ready to merge?

File: salt/pillar/git_pillar.py [hunk 1]
```diff
@@ -319,4 +314,4 @@ def ext_pillar(minion_id,
 
     pil = Pillar(opts, __grains__, minion_id, branch)
 
-    return pil.compile_pillar()
+    return pil.compile_pillar(ext=False)
```
Comment (jfindlay) [inline salt/pillar/git_pillar.py]: This change alone is causing these [test](https://jenkins.saltstack.com/job/salt-pr-rs-cent7-n/2954/testReport/junit/integration.modules.pillar/PillarModuleTest/test_ext_pillar_env_mapping/) [failures](https://jenkins.saltstack.com/job/salt-pr-rs-ubuntu14.04-n/2954/testReport/junit/integration.modules.pillar/PillarModuleTest/test_ext_pillar_env_mapping/).  I'm not sure what is going on here.  Perhaps @rallytime or @cachedout know more than I do.

**Comment by gracinet** [comment]
Not ready yet : I'd like to understand more about pillar_dirs and I'm puzzled with thé fact that this kwarg is not used elsewhere. 

Maybe passing the pillar instance and providing a method instead would ne cleaner / more efficient, but that coule be too much of a change ?

I'll take a further look today or tomorrow

On 24 mars 2015 04:06:38 UTC+01:00, jfindlay <notifications@github.com> wrote:
>@gracinet, my guess is that it is an oversight.  Do you consider this
>ready to merge?
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/saltstack/salt/pull/21680#issuecomment-85319167

-- 
Sent from my handheld device with K-9 Mail. Please excuse my brevity.

File: salt/pillar/git_pillar.py (hunk 1 above)
Comment (isbm) [inline salt/pillar/git_pillar.py]: .oO(Who tests tests?)

Comment (jfindlay) [inline salt/pillar/git_pillar.py]: @isbm, I do. :-)

**Comment by thatch45** [comment]
If we don't hear back on this one in a few days I am going to close it out.

**Comment by thatch45** [comment]
If this comes back to life please feel free to re-open it

**Comment by gracinet** [comment]
On 04/13/2015 11:45 PM, Thomas S Hatch wrote:
>
> If this comes back to life please feel free to re-open it
>
Yup sorry, got other concerns of late.
